File: python/dynamfibn.py
#!/usr/bin/env python
# File: dynamfibn.py
# Name: D.Saravanan
# Date: 29/10/2022

""" Script to compute nth fibonacci using dynamic programming """


def fibonacci(n: int) -> int:
    """compute nth fibonacci"""

    fib: list[int] = [0, 1, 1]

    for _ in range(1, n):
        fib[2] = fib[0] + fib[1]
        fib[0], fib[1] = fib[1], fib[2]

    return fib[0] if n == 0 else fib[1] if n == 1 else fib[2]


if __name__ == "__main__":
    N: int = 10
    print(fibonacci(N))

File: julia/dynamfibn.jl
#!/usr/bin/env julia
# File: dynamfibn.jl
# Name: D.Saravanan
# Date: 29/10/2022

""" Script to compute nth fibonacci using dynamic programming """


function fibonacci(n::Int)::BigInt
    """compute nth fibonacci"""

    fib::Vector{BigInt} = [0, 1, 1]

    for _ = 2:n
        fib[3] = fib[1] + fib[2]
        fib[1], fib[2] = fib[2], fib[3]
    end

    return n == 0 ? fib[1] : n == 1 ? fib[2] : fib[3]
end


function main()
    N::Int = 10000
    println(fibonacci(N))
end

main()
